assembl/views/api/post.py

```diff
@@ -168,6 +168,66 @@ def get_posts(request):
         ideaContentLinkQuery = ideaContentLinkQuery.filter(
             PostClass.hidden==asbool(hidden))
 
+
+    # "true" means deleted only, "false" (default) means non-deleted only. "any" means both.
+
+    # v0
+    # deleted = request.GET.get('deleted', 'any')
+    # end v0
+
+    # v1: we would like something like that
+    # deleted = request.GET.get('deleted', None)
+    # if deleted is None:
+    #     if view_def == 'id_only':
+    #         deleted = 'any'
+    #     else:
+    #         deleted = 'false'
+    # end v1
+
+    # v2
+    # deleted = request.GET.get('deleted', None)
+    # if deleted is None:
+    #     if not ids:
+    #         deleted = 'false'
+    #     else:
+    #         deleted = 'any'
+    #
+    # if deleted == 'false':
+    #     posts = posts.filter(PostClass.tombstone_condition())
+    #     ideaContentLinkQuery = ideaContentLinkQuery.filter(PostClass.tombstone_condition())
+    # elif deleted == 'true':
+    #     posts = posts.filter(PostClass.not_tombstone_condition())
+    #     ideaContentLinkQuery = ideaContentLinkQuery.filter(PostClass.not_tombstone_condition())
+    # elif deleted == 'any':
+    #     # result will contain deleted and non-deleted posts
+    #     pass
+    # end v2
+
+
+    # v3
+    # deleted = request.GET.get('deleted', None)
+    # if deleted is None:
+    #     if not ids:
+    #         deleted = 'false'
+    #     else:
+    #         deleted = 'any'
+
+    # if deleted == 'true':
+    #     posts = posts.filter(PostClass.not_tombstone_condition())
+    #     ideaContentLinkQuery = ideaContentLinkQuery.filter(PostClass.not_tombstone_condition())
+    # end v3
+
+    # v4
+    deleted = request.GET.get('deleted', None)
+    if deleted is None:
+        if not ids:
+            deleted = 'false'
```

Comment:
Again, `enum` types. Any time we use strings, maintainability becomes a hassle. Make a `dict` type that contains these strings, document their usages/cases. And call the dict keys instead.